### DISCUSSION TITLE
'camera-change' event does not fire while auto-rotating

### DISCUSSION THREAD

**Discussion by jukibom** [google/model-viewer#3834] [Q&A] [answered]
### Description

Using the `camera-change` event (`modelViewer.addEventListener('camera-change', () => { ... });`) does not fire when the model is rotating automatically using `auto-rotate` attribute. 

Note this DOES work with the `interaction-prompt` attribute.

#### Live Demo

https://glitch.com/edit/#!/cuddly-palm-cello

(Note you will need to open the browser console)
Console.log "Camera changed!" does not fire when auto-rotation kicks in. `interaction-prompt` has been disabled here as this already works correctly.

### Version

- model-viewer: v2.0.1

### Browser Affected

- [x] Chrome, version: 105.0.5195.127
- [x] Edge
- [x] Firefox
- [x] IE
- [x] Safari

### OS

- [x] Android
- [x] iOS
- [x] Linux
- [x] MacOS
- [x] Windows

### AR

- [x] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment, marked as answer]
Actually, this is expected behavior. The camera isn't moving during auto-rotate, the model is. It becomes clear if you add a skybox. You can find the current rotation angle through https://modelviewer.dev/docs/#entrydocs-stagingandcameras-properties-turntableRotation

**Reply by jukibom** [reply]
Ah, interesting! So for doing anything requiring syncing something to positions on the scene on the canvas, is `requestAnimationFrame` a more suitable approach?

**Reply by elalish** [reply]
If you want it based on the turntable rotation, then yes. Of course our annotations already account for this, if you want that kind of syncing.
